In [3]:
import numpy as np
import pandas as pd
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

In [4]:
filename = "../../20200404_LWB_Client_Events_Month.csv"

In [5]:
fullData = pd.read_csv(filename)

/Users/chenqinzhang/opt/anaconda3/envs/sem1_2020/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3063: DtypeWarning: Columns (849,856,934,935,936,938,939,940,941,942,943,944,945,947,948,949,951,952,953,975,976,978,1008,1014,1029) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


### Get Categories

Divide categoricals by the first starting word

In [12]:
dataCateg = []
for i in fullData.columns:
    categ = i.split("_")[0] + "_"
    if categ not in dataCateg:
        dataCateg.append(categ)

In [13]:
print(dataCateg)

['_', 'occurreddate_', 'incident_', 'rowtype_', 'rp_', 'restrictivepractice_', 'diagnosistype_', 'diagnosiscat_', 'prgcontct_', 'progsubjcat_', 'medc_', 'client_', 'allrows_', 'status_']


For example, search category start with 'occurreddate_' in data, we could simply use 
`fullData.filter(regex=r'^occurreddate_', axis=1)`.

In [15]:
categ_dict = {}
for i in dataCateg:
    if i not in categ_dict:
        categ_dict[i] = fullData.filter(regex=r'^'+i, axis=1)

In [26]:

count = 0 
for i in categ_dict:
    print("type [" + i + "] has " + str(len(categ_dict[i].columns)) + " columns")

    count += len(categ_dict[i].columns)
    
print(count)

type [_] has 5 columns
type [occurreddate_] has 5 columns
type [incident_] has 375 columns
type [rowtype_] has 160 columns
type [rp_] has 88 columns
type [restrictivepractice_] has 8 columns
type [diagnosistype_] has 16 columns
type [diagnosiscat_] has 44 columns
type [prgcontct_] has 56 columns
type [progsubjcat_] has 152 columns
type [medc_] has 23 columns
type [client_] has 12 columns
type [allrows_] has 231 columns
type [status_] has 6 columns
1181


In [19]:
print(len(fullData.columns))

1181


No data has been missed, Great!

(1).
type [_] has 5 columns

(2).
type [occurreddate_] has 5 columns

(3). (两个人)
type [incident_] has 375 columns

(4). （很多column是lags， column数折半）
type [rowtype_] has 160 columns

(5).
type [rp_] has 88 columns
type [restrictivepractice_] has 8 columns

(6).
type [diagnosistype_] has 16 columns
type [diagnosiscat_] has 44 columns

(7).
type [prgcontct_] has 56 columns
type [progsubjcat_] has 152 columns

(8).
type [medc_] has 23 columns

(9).
type [client_] has 12 columns

(10).
type [allrows_] has 231 columns

(11).
type [status_] has 6 columns




In [32]:
pd.options.display.max_seq_items = 2000
fullData.columns[:238]

Index(['_key_client_id', '_key_occurreddate_month', 'occurreddate_monthofyear',
       'incident_catgry__piv',
       'incident_catgry_offences_significant_sa_jurisdiction_cate_piv',
       'incident_catgry_client_missing_piv',
       'incident_catgry_offences_minor_niss_jurisdiction_category_piv',
       'incident_catgry_offences_minor_vic_jurisdiction_category_piv',
       'incident_catgry_offences_minor_client_missing_piv',
       'incident_catgry_offences_significant_offences_minor_piv',
       'incident_catgry_offences_minor_offences_minor_wa_jur_piv',
       'incident_catgry_behaviour_minor_behaviour_minor_piv',
       'incident_catgry_offences_minor_wa_jurisdiction_category_with_piv',
       'incident_catgry_offences_minor_offences_minor_qld_ju_piv',
       'incident_catgry_restrictive_practices_piv',
       'incident_catgry_offences_minor_qld_jurisdiction_category_with_piv',
       'incident_catgry_drug_use_and_alcohol_significant_nsw_jurisdic_piv',
       'incident_catgry_medi

In [33]:
fullData.columns[478:730]

Index(['rp_type_detail_rp_start_mechanical_restraint_piv_sum_n_days',
       'rp_type_detail_rp_end_physical_restraint_prohibited_in_cyf_piv_sum_n_days',
       'rp_type_detail_rp_start_other_piv_sum_n_days',
       'rp_type_detail_rp_end_other_piv_sum_n_days',
       'rp_type_detail_rp_start_physical_restraint_prohibited_in_cyf_piv_sum_n_days',
       'rp_type_detail_rp_start_seclusion_over_18yrs_prohibited_in_cyf_piv_sum_n_days',
       'rp_type_detail_rp_end_environmental_restraint_piv_sum_n_days',
       'rp_type_detail_rp_end_chemical_restraint_psychotropic_medication_piv_sum_n_days',
       'rp_type_detail_rp_end_restricted_access_piv_sum_n_days',
       'rp_type_detail_rp_start_restricted_access_piv_sum_n_days',
       'rp_type_detail_rp_end_mechanical_restraint_piv_sum_n_days',
       'rp_type_detail_rp_start_chemical_restraint_psychotropic_medication_piv_sum_n_days',
       'rp_type_detail_rp_start_environmental_restraint_piv_sum_n_days',
       'rp_type_detail_rp_end_exclusio

In [34]:
fullData.columns[730: 944]

Index(['rowtype_client_address_addition_piv_days_since',
       'rowtype_rp_end_piv_days_since', 'rowtype_rp_start_piv_days_since',
       'rowtype_progress_note_piv_days_since',
       'rowtype_diagnosis_piv_days_since',
       'rowtype_medictn_start_piv_days_since',
       'rowtype_client_plan_start_piv_days_since',
       'rowtype_client_plan_end_piv_days_since',
       'rowtype_keyworker_change_piv_days_since',
       'rowtype_reference_piv_days_since',
       'rowtype_medictn_end_piv_days_since',
       'rowtype_plcment_end_piv_days_since', 'rowtype_incident_piv_days_since',
       'rowtype_plcment_start_piv_days_since',
       'rowtype_program_primary_service_type_start_piv_days_since',
       'rowtype_client_alert_start_piv_days_since',
       'rp_environmental_restraint_piv_days_since',
       'rp_restricted_access_piv_days_since',
       'rp_mechanical_restraint_piv_days_since',
       'rp_exclusionary_time_out_not_including_seclusion_piv_days_since',
       'rp_physical_restr

In [35]:
fullData.columns[944:]

Index(['allrows_keyworker_min_alltime_postal_suburb_anon_hash',
       'allrows_keyworker_min_alltime_postal_state_anon_hash',
       'allrows_keyworker_min_alltime_postal_post_code',
       'allrows_keyworker_min_alltime_postal_country',
       'allrows_keyworker_min_alltime_org_level_3_key',
       'allrows_keyworker_min_alltime_org_level_4_key',
       'allrows_keyworker_min_alltime_org_level_5_key',
       'allrows_keyworker_min_alltime_employment_type_key',
       'allrows_keyworker_min_alltime_employment_type_name',
       'allrows_keyworker_min_alltime_pay_rate_group',
       'allrows_keyworker_min_in_month_award',
       'allrows_keyworker_min_in_month_position_key',
       'allrows_keyworker_min_in_month_hours_per_week',
       'allrows_keyworker_min_in_month_cost_centre_key',
       'allrows_keyworker_min_in_month_parentkey',
       'allrows_keyworker_min_in_month_gender',
       'allrows_keyworker_min_in_month_person_type_key',
       'allrows_keyworker_min_in_month_postal_s